packages/components/src/internal/components/editable/EditableGrid.tsx

```diff
@@ -545,12 +609,29 @@ export class EditableGrid extends PureComponent<EditableGridProps, EditableGridS
                         maxRowIdx = Math.max(maxRowIdx, maxInitialCell.rowIdx);
                     }
 
-                    for (let c = minColIdx; c <= maxColIdx; c++) {
-                        for (let r = minRowIdx; r <= maxRowIdx; r++) {
-                            selectionCells.push(genCellKey(c, r));
-                        }
-                    }
+                    selectionCells = computeSelectionCellKeys(minColIdx, minRowIdx, maxColIdx, maxRowIdx);
+                }
+                break;
+            case SELECTION_TYPES.AREA_CHANGE:
```

Comment:
nit: If you wrap the `case` statement body in `{}` it will scope those variables as desired (and IntellIj will no longer complain). Example:

```ts
case SELECTION_TYPES.AREA_CHANGE: {
    // body
}
```